cryptolib/verifier.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cryptolib
+
+import "errors"
+
+// Verifier contains methods to validate an Attestation.
+type Verifier interface {
+	// VerifyAttestation verifies whether an Attestation satisfies at least one
+	// of the public keys under an image. This function finds the public key
+	// whose ID matches the attestation's PublicKeyID, and uses this key to
+	// verify the signature.
+	VerifyAttestation(att *Attestation, publicKeySet []PublicKey, image string) error
+}
+
+// PublicKey stores public key material for all key types.
+type PublicKey struct {
+	// KeyData holds the raw key material which can verify a signature.
+	KeyData []byte
+	// ID uniquely identifies this public key. For PGP, this should be the
+	// OpenPGP RFC4880 V4 fingerprint of the key.
+	ID string
+}
+
+// NewPublicKey creates a new PublicKey.
+func NewPublicKey(keyData []byte, keyID string) PublicKey {
+	return PublicKey{
+		KeyData: keyData,
+		ID:      keyID,
+	}
+}
+
+type verifier struct{}
+
+// NewVerifier creates a Verifier interface for verifying Attestations.
+func NewVerifier() Verifier {
+	return &verifier{}
+}
+
+// VerifyAttestation verifies an Attestation. See Verifier for more details.
+func (v *verifier) VerifyAttestation(att *Attestation, publicKeySet []PublicKey, imageDigest string) error {
```

Comment:
As discussed in standup, make publicKeySet part of verifier. 
Also clarify imageDigest is trusted information and should not be just pulled off the attestation.

Reply:
Done.